lib/rubocop/cop/variable_force/assignment.rb

```diff
@@ -119,7 +120,15 @@ def rest_assignment_node
         end
 
         def for_assignment_node
-          node.ancestors.find(&:for_type?)
+          parent_node = node.parent
+          return unless parent_node
+
+          grandparent_node = parent_node.parent
+
+          return parent_node if parent_node.for_type?
+          return grandparent_node if parent_node.mlhs_type? && grandparent_node&.for_type?
+
+          nil
```

Comment:
Can you tweak to the following? There is a possibility for an early return before searching for `grandparent_node`.

```suggestion
          return unless (parent_node = node.parent)
          return parent_node if parent_node.for_type?

          grandparent_node = parent_node.parent
          return grandparent_node if parent_node.mlhs_type? && grandparent_node&.for_type?

          nil
```

Reply:
Updated.